app/apps/handlers.js

```diff
@@ -50,10 +49,14 @@ exports.list = (req, res, next) => {
 exports.details = (req, res, next) => {
   Promise.all([App.get(req.params.id), Bucket.list(), User.list()])
     .then(([app, buckets, users]) => {
+      const user_ids = app.data.userapps.map(userapp => userapp.user.auth0_id);
```

Comment:
As discussed in Slack, we currently don't use the UserApp relationship for permission-checking but we may start at some point.

That relationship also has an `is_admin` field which should potentially give you not only access to an app but also ability to perform potentially destructive operations.

That field is ignored in this case despite the variable name called `current_user_is_app_admin`, this may be alright for now but it may require tweaking in the future.

Reply:
e.g. a user whose `UserApp` record has the `is_admin` flag set to `false` can see the buttons to add/remove "user<->app" relationships.

Reply:
True. At the moment there is no difference between a user connected to an app who is "an admin" and one who isn't. When connecting a user to an app the `is_admin` field is set to true, so it's always going to be true. That may change.

Reply:
Yes. At the moment the `is_admin` flag is completely superfluous.

Reply:
Don't access `app.data.userapps`, use `app.users`, eg:
```javascript
const user_ids = app.users.map(user => user.auth0_id);
```

Reply:
Oh, there isn't one. But you can add it pretty easily in `app/models.js`. Eg:
```javascript
  get users() {
    return new ModelSet(User, this.data.userapps.map(ua => ua.user));
  }
```